lib/ansible/plugins/action/__init__.py

```diff
@@ -163,6 +163,8 @@ def _configure_module(self, module_name, module_args, task_vars):
         else:
             use_vars = task_vars
 
+        leaf_module_name = module_name.split('.')[-1]
```

Comment:
fwiw we have `ansible.utils.collection_loader.resource_from_fqcr` that does this.

Reply:
We probably should use that method since it's doing some additional text conversion to ensure it's text type.